src/profile-logic/convert-markers.js

```diff
@@ -127,3 +130,49 @@ export function upgradeGCMajorMarker_Processed8to9(
       throw new Error('Unknown GCMajor status');
   }
 }
+
+export function processIPCMarkers(threads: Array<Thread>) {
```

Comment:
I'm wondering if this work could happen in a selector instead of during profile processing. Profile processing is "forever", and if there is a bug, then it is quite hard to go back and fix. Selectors run every time you load the profile, and compute derived data.

Here is where we derive the markers:

https://github.com/firefox-devtools/profiler/blob/f9bf24485b60c9a9ccd4cfcb90f22ea129097a6d/src/selectors/per-thread/markers.js#L82-L89

It could be possible to wire in the information to this selector, or create a new one outside of the per-thread infrastructure. Happy to chat real-time about this if you want.